scripts/src/public/services/payment.ts

```diff
@@ -117,7 +121,48 @@ export type BlockchainConfig = {
 	asset_code: string;
 };
 
-export async function getBlockchainConfig(logger: LoggerInstance): Promise<BlockchainConfig> {
-	const res = await client.get(`${config.payment_service}/config`);
+export async function getBlockchainConfig(blockchainVersion: string, logger: LoggerInstance): Promise<BlockchainConfig> {
+	const res = await client.get(`${getPaymentServiceUrl(blockchainVersion)}/config`);
 	return res.data;
 }
+
+export interface WhitelistTransactionRequest {
+	order_id: string;
+	source: string;
+	destination: string;
+	amount: number;
+	xdr: string;
+	network_id: string;
+	app_id: string;
+}
+
+export async function whitelistTransaction(
+	order: db.Order,
+	network_id: string,
+	tx_envelope: string,
+	app_id: string): Promise<string> {
+
+	const payload: WhitelistTransactionRequest = {
+		order_id: order.id,
+		source: order.blockchainData.sender_address!,
+		destination: order.blockchainData.recipient_address!,
+		amount: order.amount,
+		xdr: tx_envelope,
+		network_id,
+		app_id
+	};
+
+	// whitelist is only for the new payment service
+	const whitelist_response = (await client.post(`${config.new_payment_service}/whitelist`, payload));
+	if (whitelist_response.status === 401) {
```

Comment:
I think that when a status code greater than 300 is received by `axios` it raises an error. So this flow will never enter. You need to catch the error and rethrow it if you want to throw a specific error

Reply:
Ok, changed it :+1: